airflow/models.py

```diff
@@ -103,6 +103,28 @@
 _CONTEXT_MANAGER_DAG = None
 
 
+def stats_incr_helper(stat, value, dag_id, task_id):
```

Comment:
There's an alternative approach where you decorate incr and gauge to check and see if tags contains both task_id and dag_id. If this sort of helper func gets applied to more events, we'll probably want to clean it up.... but... YAGNI. :)